internal/datanode/flow_graph_insert_buffer_node.go

```diff
@@ -382,25 +383,19 @@ func (ibNode *insertBufferNode) FillInSyncTasks(fgMsg *flowGraphMsg, seg2Upload
 	// every time we make sure that the N biggest delDataBuf can be flushed
 	// when memsize usage reaches a certain level
 	// the aim for taking all these actions is to guarantee that the memory consumed by delBuf will not exceed a limit
-	segmentsToFlush := ibNode.delBufferManager.ShouldFlushSegments()
-	for _, segID := range segmentsToFlush {
-		syncTasks[segID] = &syncTask{
-			buffer:    nil, // nil is valid
-			segmentID: segID,
-		}
-	}
+	segmentsToFlush := typeutil.NewUniqueSet(ibNode.delBufferManager.ShouldFlushSegments()...)
+	segmentsToFlush.Insert(ibNode.channel.listSegmentIDsToSync(fgMsg.endPositions[0].Timestamp)...)
```

Comment:
Synced segments do not necessarily require flushing. I thinkl we cannot set `task.flushed = true` for these segments.

Reply:
@bigsheeper `segmentsToFlush` here seem to be `segmentsToSync`, they are not set `Flushed=true`. Looks like naming issue?

Reply:
@congqixia Correct me if I'm wrong, I think they will be set at line 425:
```Golang
	mergeSyncTask(flushedSegments, syncTasks, func(task *syncTask) {
		task.flushed = true
	})
```

Reply:
@bigsheeper flushedSegments was get at line 424
```
flushedSegments := ibNode.CollectSegmentsToSync()
```
means get manual sync segment from flushCh. Will not set segmentsToFlush

Reply:
Oh, sorry, my mistake.

Reply:
@bigsheeper actually it's naming issue, shall fix this later